xds/internal/resolver/xds_resolver_test.go

```diff
@@ -83,7 +83,7 @@ var (
 	target = resolver.Target{URL: *testutils.MustParseURL("xds:///" + targetStr)}
 
 	routerHTTPFilter = httpfilter.Get(router.TypeURL)
-	routerConfig, _  = routerHTTPFilter.ParseFilterConfig(testutils.MarshalAny(&v3routerpb.Router{}))
+	routerConfig, _  = routerHTTPFilter.ParseFilterConfig(testutils.MarshalAny(&testing.T{}, &v3routerpb.Router{}))
```

Comment:
Please add a function instead of these global vars. That way you would be able to pass a valid `testing.T` to `testutils.MarshalAny`.

```golang
func makeRouterFilter(t *testing.T) xdsresource.HTTPFilter {
	f := httpfilter.Get(router.TypeURL)
	cfg, _  = f.ParseFilterConfig(testutils.MarshalAny(t, &v3routerpb.Router{}))
	return xdsresource.HTTPFilter{Name: "rtr", Filter: routerHTTPFilter, Config: cfg}
}
```